Author: Pino Toscano <pino@debian.org>
Description: Adjust the installation paths to the standard Debian locations.
 .
 'include(QtCreatorAPI)' is moved after 'project()', otherwise the compiler
 is not yet detected and thus ${CMAKE_LIBRARY_ARCHITECTURE} is not defined.
 .
 See also https://bugreports.qt.io/browse/QTCREATORBUG-25142
Forwarded: not-needed
Last-Update: 2020-12-18

--- a/cmake/QtCreatorAPIInternal.cmake
+++ b/cmake/QtCreatorAPIInternal.cmake
@@ -51,20 +51,20 @@ else ()
   set(_IDE_APP_PATH "bin")
   set(_IDE_APP_TARGET "${IDE_ID}")
 
-  set(_IDE_LIBRARY_BASE_PATH "lib")
-  set(_IDE_LIBRARY_PATH "lib/qtcreator")
-  set(_IDE_PLUGIN_PATH "lib/qtcreator/plugins")
+  set(_IDE_LIBRARY_BASE_PATH "lib/${CMAKE_LIBRARY_ARCHITECTURE}")
+  set(_IDE_LIBRARY_PATH "lib/${CMAKE_LIBRARY_ARCHITECTURE}/qtcreator")
+  set(_IDE_PLUGIN_PATH "lib/${CMAKE_LIBRARY_ARCHITECTURE}/qtcreator/plugins")
   if (WIN32)
     set(_IDE_LIBEXEC_PATH "bin")
     set(QT_DEST_PLUGIN_PATH "bin/plugins")
     set(QT_DEST_QML_PATH "bin/qml")
   else ()
     set(_IDE_LIBEXEC_PATH "libexec/qtcreator")
-    set(QT_DEST_PLUGIN_PATH  "lib/Qt/plugins")
-    set(QT_DEST_QML_PATH "lib/Qt/qml")
+    set(QT_DEST_PLUGIN_PATH  "lib/${CMAKE_LIBRARY_ARCHITECTURE}/Qt/plugins")
+    set(QT_DEST_QML_PATH "lib/${CMAKE_LIBRARY_ARCHITECTURE}/Qt/qml")
   endif ()
   set(_IDE_DATA_PATH "share/qtcreator")
-  set(_IDE_DOC_PATH "share/doc/qtcreator")
+  set(_IDE_DOC_PATH "share/qtcreator/doc")
   set(_IDE_BIN_PATH "bin")
 endif ()
 
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,7 +7,6 @@ include(FeatureSummary)
 include(QtCreatorIDEBranding RESULT_VARIABLE IDE_BRANDING_FILE)
 include(QtCreatorTranslations)
 include(QtCreatorDocumentation)
-include(QtCreatorAPI)
 
 set(IDE_REVISION FALSE CACHE BOOL "Marks the presence of IDE revision string.")
 set(IDE_REVISION_STR "" CACHE STRING "The IDE revision string.")
@@ -17,6 +16,8 @@ mark_as_advanced(IDE_REVISION IDE_REVISI
 
 project(QtCreator VERSION ${IDE_VERSION})
 
+include(QtCreatorAPI)
+
 # Force C++ standard, do not fall back, do not use compiler extensions
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
